java/kotlin-extractor/src/main/kotlin/KotlinFileExtractor.kt

```diff
@@ -5290,48 +5296,44 @@ open class KotlinFileExtractor(
                 }
             }
         }
     }
 
     /**
      * Extracts a type access expression and its child type access expressions in case of a generic type. Nested generics are also handled.
-     * No enclosing callable and statement is extracted, this is useful for type access extraction in field declarations.
+     * No enclosing callable and statement is extracted; this is useful for type access extraction in field declarations.
      */
     private fun extractTypeAccessRecursive(t: IrType, location: Label<out DbLocation>, parent: Label<out DbExprparent>, idx: Int, typeContext: TypeContext = TypeContext.OTHER): Label<out DbExpr> {
-        val typeAccessId = extractTypeAccess(useType(t, typeContext), location, parent, idx)
-        if (t is IrSimpleType) {
-            t.arguments.forEachIndexed { argIdx, arg ->
-                extractWildcardTypeAccessRecursive(arg, location, typeAccessId, argIdx)
-            }
-        }
-        return typeAccessId
+        return extractTypeAccessRecursive(t, location, parent, idx, null, null, typeContext)
     }
 
     /**
      * Extracts a type access expression and its child type access expressions in case of a generic type. Nested generics are also handled.
      */
-    private fun extractTypeAccessRecursive(t: IrType, location: Label<DbLocation>, parent: Label<out DbExprparent>, idx: Int, enclosingCallable: Label<out DbCallable>?, enclosingStmt: Label<out DbStmt>?, typeContext: TypeContext = TypeContext.OTHER, overrideId: Label<out DbExpr>? = null): Label<out DbExpr> {
+    private fun extractTypeAccessRecursive(t: IrType, location: Label<out DbLocation>, parent: Label<out DbExprparent>, idx: Int, enclosingCallable: Label<out DbCallable>?, enclosingStmt: Label<out DbStmt>?, typeContext: TypeContext = TypeContext.OTHER, overrideId: Label<out DbExpr>? = null): Label<out DbExpr> {
         // TODO: `useType` substitutes types to their java equivalent, and sometimes that also means changing the number of type arguments. The below logic doesn't take this into account.
         // For example `KFunction2<Int,Double,String>` becomes `KFunction<String>` with three child type access expressions: `Int`, `Double`, `String`.
         val typeAccessId = extractTypeAccess(useType(t, typeContext), location, parent, idx, enclosingCallable, enclosingStmt, overrideId = overrideId)
         if (t is IrSimpleType) {
             if (t.arguments.isNotEmpty() && overrideId != null) {
                 logger.error("Unexpected parameterized type with an overridden expression ID; children will be assigned fresh IDs")
             }
-            extractTypeArguments(t.arguments.filterIsInstance<IrType>(), location, typeAccessId, enclosingCallable, enclosingStmt)
+            t.arguments.forEachIndexed { argIdx, arg ->
+                extractWildcardTypeAccessRecursive(arg, location, typeAccessId, argIdx, enclosingCallable, enclosingStmt)
+            }
```

Comment:
This now does roughly what the other `extractTypeAccessRecursive` function used to do.